lang/java/reef-applications/reef-vortex/src/test/java/org/apache/reef/vortex/driver/DefaultVortexMasterTest.java

```diff
@@ -157,6 +159,59 @@ public void testMultipleTaskletsFailure() throws Exception {
   }
 
   /**
+   * Test handling of single tasklet execution with a cancellation after launch.
+   */
+  @Test(timeout = 10000)
+  public void testSingleTaskletCancellation() throws Exception {
+    final RunningWorkers runningWorkers = new RunningWorkers(new RandomSchedulingPolicy());
+    final PendingTasklets pendingTasklets = new PendingTasklets();
+    final VortexFuture future = createTaskletCancellationFuture(runningWorkers, pendingTasklets);
+    launchTasklets(runningWorkers, pendingTasklets, 1);
+
+    assertTrue(future.cancel(true));
+    assertTrue("TheVortexFuture should be cancelled.", future.isCancelled());
+    assertTrue("The VortexFuture should be done", future.isDone());
+  }
+
+  /**
+   * Test handling of single tasklet execution with a cancellation before launch.
+   */
+  @Test(timeout = 10000)
+  public void testSingleTaskletCancellationBeforeLaunch() throws Exception {
+
+    final RunningWorkers runningWorkers = new RunningWorkers(new RandomSchedulingPolicy());
+    final PendingTasklets pendingTasklets = new PendingTasklets();
+    final VortexFuture future = createTaskletCancellationFuture(runningWorkers, pendingTasklets);
+
+    Exception exception = null;
+    try {
+      future.cancel(true, 100, TimeUnit.MILLISECONDS);
```

Comment:
Having `fail()` here would be simpler than assigning an exception and checking type of it.